pkg/kubelet/kuberuntime/security_context.go

```diff
@@ -0,0 +1,150 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kuberuntime
+
+import (
+	"k8s.io/kubernetes/pkg/api"
+	runtimeapi "k8s.io/kubernetes/pkg/kubelet/api/v1alpha1/runtime"
+)
+
+// determineEffectiveSecurityContext gets container's security context from api.Pod and container.SecurityContext.
+func (m *kubeGenericRuntimeManager) determineEffectiveSecurityContext(pod *api.Pod, sc *api.SecurityContext) *runtimeapi.SecurityContext {
```

Comment:
Another reason is `determineEffectiveSecurityContext ` returns `runtimeapi.SecurityContext` instead of `api.SecurityContext`, and `runtimeapi.SecurityContext` contains more fields than `api.SecurityContext`. 

Will change to `DetermineEffectiveSecurityContext()`.